File: app/src/main/java/ai/kortnevdmitriy/msafiri/adapters/TicketsAdapter.java
package ai.kortnevdmitriy.msafiri.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import ai.kortnevdmitriy.msafiri.R;
import ai.kortnevdmitriy.msafiri.entities.TicketDetails;

/**
 * Created by kortn on 11/22/2017.
 */

public class TicketsAdapter extends RecyclerView.Adapter<TicketsAdapter.TicketsViewHolder> {
    private List<TicketDetails> ticketDetails;
    private OnItemClickListener listener;

    public TicketsAdapter(List<TicketDetails> ticketDetails, OnItemClickListener listener) {
        this.ticketDetails = ticketDetails;
        this.listener = listener;
    }

    @Override
    public TicketsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.tickets_list_row, parent, false);

        return new TicketsAdapter.TicketsViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(TicketsViewHolder holder, int position) {
        TicketDetails detailsOfTickets = ticketDetails.get(position);
        holder.companyName.setText(detailsOfTickets.getCompanyName());
        holder.vehicleType.setText(detailsOfTickets.getBoardingPoint());
        holder.travelRoute.setText(detailsOfTickets.getTravelRoute());
        holder.priceInKsh.setText(detailsOfTickets.getPriceInKsh());
        holder.numberPlate.setText(detailsOfTickets.getNumberPlate());
        holder.seatNumber.setText(detailsOfTickets.getSeatNumber());
        holder.bind(detailsOfTickets, listener);
    }

    @Override
    public int getItemCount() {
        return ticketDetails.size();
    }

    public interface OnItemClickListener {
        void onItemClick(TicketDetails item);
    }

    public class TicketsViewHolder extends RecyclerView.ViewHolder {
        TextView companyName, travelRoute, vehicleType, priceInKsh, numberPlate, seatNumber;
        String key;

        public TicketsViewHolder(View view) {
            super(view);
            companyName = view.findViewById(R.id.ticketsCompanyName);
            vehicleType = view.findViewById(R.id.ticketsVehicleType);
            travelRoute = view.findViewById(R.id.ticketsTravelRoute);
            priceInKsh = view.findViewById(R.id.ticketsPriceInKsh);
            numberPlate = view.findViewById(R.id.ticketsNumberPlate);
            seatNumber = view.findViewById(R.id.ticketsSeatNumber);
        }

        void bind(final TicketDetails item, final OnItemClickListener listener) {
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    listener.onItemClick(item);
                }
            });
        }
    }
}

File: app/src/main/java/ai/kortnevdmitriy/msafiri/adapters/TicketsAdapter.kt
package ai.kortnevdmitriy.msafiri.adapters

import ai.kortnevdmitriy.msafiri.R
import ai.kortnevdmitriy.msafiri.models.TicketDetails
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by kortn on 11/22/2017.
 */

class TicketsAdapter(
	private val ticketDetails: List<TicketDetails>,
	private val listener: OnItemClickListener
) : RecyclerView.Adapter<TicketsAdapter.TicketsViewHolder>() {
	
	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TicketsViewHolder {
		val itemView = LayoutInflater.from(parent.context)
			.inflate(R.layout.tickets_list_row, parent, false)
		
		return TicketsAdapter.TicketsViewHolder(itemView)
	}
	
	override fun onBindViewHolder(holder: TicketsViewHolder, position: Int) {
		val detailsOfTickets = ticketDetails[position]
		holder.companyName.text = detailsOfTickets.companyName
		holder.vehicleType.text = detailsOfTickets.boardingPoint
		holder.travelRoute.text = detailsOfTickets.travelRoute
		holder.priceInKsh.text = detailsOfTickets.priceInKsh
		holder.numberPlate.text = detailsOfTickets.numberPlate
		holder.seatNumber.text = detailsOfTickets.seatNumber
		holder.bind(detailsOfTickets, listener)
	}
	
	override fun getItemCount(): Int {
		return ticketDetails.size
	}
	
	interface OnItemClickListener {
		fun onItemClick(item: TicketDetails)
	}
	
	class TicketsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
		internal var companyName: TextView = view.findViewById(R.id.ticketsCompanyName)
		internal var travelRoute: TextView = view.findViewById(R.id.ticketsTravelRoute)
		internal var vehicleType: TextView = view.findViewById(R.id.ticketsVehicleType)
		internal var priceInKsh: TextView = view.findViewById(R.id.ticketsPriceInKsh)
		internal var numberPlate: TextView = view.findViewById(R.id.ticketsNumberPlate)
		internal var seatNumber: TextView = view.findViewById(R.id.ticketsSeatNumber)
		internal var key: String? = null
		
		internal fun bind(item: TicketDetails, listener: OnItemClickListener) {
			itemView.setOnClickListener { listener.onItemClick(item) }
		}
	}
}
